src/core/device.c

```diff
@@ -132,6 +291,23 @@ static int device_load(Unit *u) {
                         log_unit_debug_errno(u, r, "Failed to unescape name: %m");
         }
 
+        if (!verify_io_cost_supported(d)) {
+                log_debug_errno(SYNTHETIC_ERRNO(EOPNOTSUPP), "IO cost is not supported");
+                return 0;
+        }
```

Comment:
nah, this should not fail/be skipped. We should handle this gracefully, and simply not apply the iocost stuff in that case.